onnxruntime/core/providers/cpu/ml/label_encoder.h

```diff
@@ -105,5 +106,80 @@ class LabelEncoder_2 final : public OpKernel {
   // ONNX attribute name to load values.
   std::string _value_field_name;
 };
+
+template <typename T>
+std::vector<T> GetAttribute(const OpKernelInfo& info, const std::string& name, const std::string& tensor_name) {
+  std::vector<T> attrs;
+  auto result = info.GetAttrs<T>(name, attrs);
+  if (!result.IsOK()) {
+    ONNX_NAMESPACE::TensorProto attr_tensor_proto;
+    result = info.GetAttr(tensor_name, &attr_tensor_proto);
+    ORT_ENFORCE(result.IsOK(), "LabelEncoder is missing an attribute");
+    return utils::ParseData<T>(attr_tensor_proto);
+  }
+  return attrs;
+}
+
+template <typename T>
+T GetDefault(const OpKernelInfo& info, const std::string& attr_name, const T& backup) {
+  ONNX_NAMESPACE::TensorProto attr_tensor_proto;
+  auto result = info.GetAttr("default_tensor", &attr_tensor_proto);
+  if (result.IsOK() && utils::HasDataType(attr_tensor_proto)) {
+    auto default_value = utils::ParseData<T>(attr_tensor_proto);
+    ORT_ENFORCE(default_value.size() == 1, "default_tensor must have exactly one element");
+    return default_value[0];
+  } else {
+    T default_value;
+    result = info.GetAttr<T>(attr_name, &default_value);
+    if (result.IsOK()) {
+      return default_value;
+    } else {
+      return backup;
+    }
+  }
+}
+
+template <typename TKey, typename TValue>
+class LabelEncoder_4 final : public OpKernel {
+ public:
+  LabelEncoder_4(const OpKernelInfo& kernel_info) : OpKernel(kernel_info) {
+    InitializeAttrFields(kernel_info);
+    auto keys = GetAttribute<TKey>(kernel_info, _key_field_name, "keys_tensor");
+    auto values = GetAttribute<TValue>(kernel_info, _value_field_name, "values_tensor");
+    ORT_ENFORCE(keys.size() == values.size(), "Keys and values must have the same length.");
+    for (size_t i = 0; i < keys.size(); ++i) {
+      _map.emplace(keys[i], values[i]);
+    }
+  }
+  Status Compute(OpKernelContext* context) const override {
+    const auto* X = context->Input<Tensor>(0);
+    if (nullptr == X) {
+      return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+    }
+    const TensorShape& shape = X->Shape();
+    auto* Y = context->Output(0, shape);
+
+    auto input = X->template DataAsSpan<TKey>();
+    auto output = Y->template MutableDataAsSpan<TValue>();
+
+    for (int64_t i = 0; i < shape.Size(); ++i) {
+      const auto found = _map.find(input[onnxruntime::narrow<size_t>(i)]);
+      if (found == _map.end())
+        output[onnxruntime::narrow<size_t>(i)] = _default_value;
+      else
+        output[onnxruntime::narrow<size_t>(i)] = found->second;
+    }
+
+    return Status::OK();
+  }
+
+ private:
+  void InitializeAttrFields(const OpKernelInfo& kernel_info);
+  InlinedHashMapNaNSensitive<TKey, TValue> _map;
+  TValue _default_value;
+  std::string _key_field_name;
```

Comment:
Names with leading underscored are reserved. Please, use trailing underscore to designate a data member.

Reply:
Again, this was the existing code style in the file. I've updated the existing code to conform too.

Reply:
We do not want to multiply code that is not conformant.